src/Juvix/Data/Effect/Files/IO.hs

```diff
@@ -5,89 +5,34 @@ module Juvix.Data.Effect.Files.IO
 where
 
 import Data.ByteString qualified as ByteString
-import Data.HashSet qualified as HashSet
 import Juvix.Data.Effect.Files.Base
-import Juvix.Extra.Stdlib qualified as Stdlib
 import Juvix.Prelude.Base
+import Juvix.Prelude.Path
+import Path.IO qualified as Path
+import System.Posix.Types qualified as P
+import System.PosixCompat.Files qualified as P
 
--- | Try to resolve a filepath `p` to a path within standard library.
---
--- When `p` is not a member of the standard library or no stanard library is
--- registered, resolve `p` to an absolute path based at `rootPath` instead.
---
--- This function throws an error if `p` is a member of the standard library and
--- also present within `rootPath`.
-stdlibOrFile ::
-  forall r.
-  Members '[Embed IO, Error FilesError] r =>
-  FilePath ->
-  FilePath ->
-  Maybe StdlibState ->
-  Sem r FilePath
-stdlibOrFile p rootPath m = case m of
-  Nothing -> return (rootPath </> p)
-  Just s
-    | HashSet.member (normalise p) (s ^. stdlibFilePaths) ->
-        ifM
-          isConflict
-          ( throw
-              FilesError
-                { _filesErrorPath = pAbsPath,
-                  _filesErrorCause = StdlibConflict
-                }
-          )
-          (return (s ^. stdlibRoot </> p))
-    | otherwise -> return pAbsPath
-    where
-      pAbsPath :: FilePath
-      pAbsPath = rootPath </> p
-
-      isConflict :: Sem r Bool
-      isConflict = do
-        cRootPath <- embed (canonicalizePath rootPath)
-        cStdlibPath <- embed (canonicalizePath (s ^. stdlibRoot))
-        andM [return (cRootPath /= cStdlibPath), embed (doesFileExist pAbsPath)]
```

Comment:
Previously we did not consider file lookups inside the stdlib to be file conflicts (`cRootPath /= cStdlibPath`). This facilitates stdlib development and the REPL depends on this behaviour (caught in the shellcheck failtures).

```
$ cd tests/positive/Internal
$ juvix repl
/Users/paul/heliax/juvix/tests/positive/Internal/.juvix-build/stdlib/Stdlib/Prelude.juvix:1:8-22: error:
The top module Stdlib.Prelude is defined in the file:
/Users/paul/heliax/juvix/tests/positive/Internal/.juvix-build/stdlib/Stdlib/Prelude.juvix
But it should be in the file:
/Users/paul/heliax/juvix/tests/positive/Internal/Stdlib/Prelude.juvix
```

Reply:
this has been addressed